src/lib/mongoBackend/mongoUpdateContextSubscription.cpp

```diff
@@ -208,34 +207,43 @@ HttpStatusCode mongoUpdateContextSubscription
   }
   responseP->subscribeResponse.subscriptionId = requestP->subscriptionId;
 
-  reqSemGive(__FUNCTION__, "ngsi10 update subscription request", reqSemTaken);
 
 
   //
   // Modification of the subscription cache
   //
-  // The subscription before this updatye is stored in 'sub'.
-  // The updated subscription is stored in 'newSub'
+  // The subscription "before this update" is looked up in cache and referenced by 'cSubP'.
+  // The "updated subscription information" is in 'newSubObject' (mongo BSON object format).
   // 
   // All we need to do now for the cache is to:
-  //   1. Remove 'sub' from sub-cache (if present)
-  //   2. Create 'newSub' in sub-cache (if applicable)
+  //   1. Remove 'cSubP' from sub-cache (if present)
+  //   2. Create 'newSubObject' in sub-cache (if applicable)
+  //
+  // The subscription is already updated in mongo.
+  //
+  // FIXME P7: mongoSubCache stuff should be avoided if subscription is not patterned
```

Comment:
Yeah, the check is done later, unfortunately. 
In:

    mongoSubCacheItemInsert(tenant.c_str(), newSubObject, subscriptionId, servicePath);

It would be faster to do the check before, and not even call mongoSubCacheItemInsert if the subscription is not  patterned.

But also, doing the check in mongoSubCacheItemInsert, saves us from bad future uses of an 'unprotected' mongoSubCacheItemInsert.

This is the exact opposite of another insert function we discussed yesterday.